### ISSUE TITLE
Clusters on top of/layered across /extended other clusters?

### ISSUE THREAD

**Issue by kbrady-vibrato** [mingrammer/diagrams#111]
Hi,

First off - thanks for an awesome diagramming tool - its been a really good experience the last couple of days getting stuff down in my head in to code.

That said, I have ht a bit of a quandry trying to represent a scaling set across 2 Azure subnets - this would be 3 clusters ideally:

```
with Cluster("SubnetA"):
  with Cluster("Scaling Set"):
    nodes = [
      VMLinux("node"),
      VMLinux("node")
  ]

with Cluster("SubnetB"):
  <include Cluster("Scaling Set") somehow>
```

I would like the `Scaling Set` cluster to be extended across both `SubnetA` and `SubnetB` - is this possible?

Thanks for all your hard work!

**Comment by Karreg** [comment]
Hello,

I juste discovered your tool this morning, and I agree this is a great tool that deserve to be known! I've been looking for something like that for years.

I agree with this request. Another basic use case is cloud organization through multiple AZs.

A DB cluster is a cluster, but each node are in a different AZ, each AZ being clusters too...

**Comment by mingrammer** [comment]
Interesting. I'll have to review if this is possible and get back to here. Thank you.

**Comment by acere** [comment]
Amazing tool, it could easily become my daily go-to, if this feature can be implemented!

**Comment by clayms** [comment]
@kbrady-vibrato  @acere  @Karreg  
Is below along the lines of what you were thinking?

```
from diagrams import Diagram, Cluster, Edge, Node

graph_attr = {
    "layout":"neato",
    }

scaling_clus_attr = {
    "bgcolor":"transparent",
    "pencolor":"blue",
    "penwidth":"4.0"
    }

with Diagram("\n\nOverlapping Clusters", show=False, graph_attr=graph_attr) as diag:

    with Cluster("SubnetA"):
        A_UpLf = Node("", shape="plaintext", pin="true", pos="0,4")
        A_LwRt = Node("", shape="plaintext", pin="true", pos="4,0")
    with Cluster("SubnetB"):
        B_UpLf = Node("", shape="plaintext", pin="true", pos="6,4")
        B_LwRt = Node("", shape="plaintext", pin="true", pos="10,0")
    with Cluster("Scaling Set", graph_attr=scaling_clus_attr):
        B_UpLf = Node("", shape="plaintext", pin="true", pos="2,3")
        B_LwRt = Node("", shape="plaintext", pin="true", pos="8,1")
        n1 = Node("node 1", shape="circle", labelloc="c", pin="true", pos="3,2")
        n1 = Node("node 2", shape="box",    labelloc="c", pin="true", pos="7,2")


diag
```

![image](https://user-images.githubusercontent.com/12793783/98178250-4e405c80-1eca-11eb-96ab-3d74d80fafb2.png)

**Comment by kbrady-vibrato** [comment]
hi @clayms - yes this is pretty much what i envisioned

**Comment by acere** [comment]
Thanks @clayms,  this is the correct visual representation! 
 
It would be great to be able to achieve the same without having to manually pin down the cluster and the nodes, but I understand that computing the correct item position becomes quickly hard to manage.

**Comment by Karreg** [comment]
Yes, it's exactly the need. Without position computing it would be great (since any other component can move on it's own).
